File: Python/Level 1/3진법 뒤집기.py
def solution(n):
    answer = 0
    li = []
    
    while (n > 0):
        x = n%3
        n = (n-x)//3
        li.append(x)
    
    for i in range(len(li)):
        sq = len(li) - i - 1
        answer += (3 ** sq) * li[i]
    
    return answer

File: Java/Level 1/3진법 뒤집기.java
import java.util.LinkedList;

public class Digit3 {
    public int solution(int n) {
        int digitNum = 3;
        LinkedList<Integer> convertedNum = new LinkedList<>();

        while (n > 0) {
            convertedNum.addFirst(n % digitNum);
            n /= digitNum;
        }

        return convertedNum.stream().reduce(0, (acc, cur) -> acc + cur * (int)Math.pow(digitNum, convertedNum.indexOf(cur)));
    }

    public int answer(int n) {
        int result = 0;
        while (n > 0) {
            result = (n % 3) + (result * 3);
            n /= 3;
        }

        return result;
    }
}
